sdk/src/main/java/com/ingresse/sdk/services/TransferService.kt

```diff
@@ -0,0 +1,256 @@
+import retrofit2.Call
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+import retrofit2.converter.scalars.ScalarsConverterFactory
+import java.io.IOException
+
+class TransferService(private val client: IngresseClient) {
+    private var host = Host.API
+    private var service: Transfer
+    private var cancelAllCalled = false
+
+    private var mUserTransfersCall: Call<String>? = null
+    private var mRecentTransfersCall: Call<String>? = null
+    private var mFriendsFromSearchCall: Call<String>? = null
+    private var mConcurrentCalls: ArrayList<Call<String>> = ArrayList()
+
+    init {
+        val httpClient = ClientBuilder(client)
+            .addRequestHeaders()
+            .build()
+
+        val adapter = Retrofit.Builder()
+            .addConverterFactory(ScalarsConverterFactory.create())
+            .addConverterFactory(GsonConverterFactory.create())
+            .client(httpClient)
+            .baseUrl(URLBuilder(host, client.environment).build())
+            .build()
+
+        service = adapter.create(Transfer::class.java)
+    }
+
+    /**
+     * Method to cancel all requests
+     */
+    fun cancelAll() {
+        cancelAllCalled = true
+        mUserTransfersCall?.cancel()
+        mRecentTransfersCall?.cancel()
+        mFriendsFromSearchCall?.cancel()
+        mConcurrentCalls.forEach { it.cancel() }
+        mConcurrentCalls.clear()
+    }
+
+    /**
+     * Method to cancel user transfers data request
+     */
+    fun cancelUserTransfersData(concurrent: Boolean = false) {
+        if(!concurrent) {
+            mUserTransfersCall?.cancel()
+            return
+        }
+
+        mConcurrentCalls.forEach { it.cancel() }
+        mConcurrentCalls.clear()
+    }
+
+    /**
+     * Method to cancel recent transfers data request
+     */
+    fun cancelRecentTransfersData(concurrent: Boolean = false) {
+        if(!concurrent) {
+            mRecentTransfersCall?.cancel()
+            return
+        }
+
+        mConcurrentCalls.forEach { it.cancel() }
+        mConcurrentCalls.clear()
+    }
+
+    /**
+     * Method to cancel a get friends from search
+     */
+    fun cancelFriendsFromSearch(concurrent: Boolean = false) {
+        if (!concurrent) {
+            mFriendsFromSearchCall?.cancel()
+            return
+        }
+
+        mConcurrentCalls.forEach { it.cancel() }
+        mConcurrentCalls.clear()
+    }
+
+    /**
+     * Get user transfers data
+     *
+     * @param concurrent - parameters to concurrent request
+     * @param request - parameters required to request
+     * @param onSuccess - success callback
+     * @param onError - error callback
+     * @param onConnectionError - connection error callback
+     */
+    fun getUserTransfersData(concurrent: Boolean = false,
+                             request: UserTransfersData,
+                             onSuccess: (com.ingresse.sdk.base.Array<UserTransfersJSON>) -> Unit,
```

Comment:
See if is possible remove the complete name and put in import, and only use `Array<UserTransferJSON>`